src/gui/mzroll/background_peaks_update.cpp

```diff
@@ -320,7 +320,7 @@ void BackgroundPeakUpdate::run(void) {
                 quit();
                 return;
         }
-
+        connect(this,SIGNAL(alignmentError(QString)),mainwindow->alignmentErrorDialog,SLOT(showMessage(QString)));
```

Comment:
Why are we asking MainWindow to show an error message related to alignment ? Maybe alignment dialog should handle this. ? 
What do you think ?

Reply:
@rish9511 background_peaks_update.cpp has all kind of job getting done in different thread from GUI thread and that(GUI) is handled by mainwindow.cpp. This architecture make whole ElMaven more responsive.